### PR TITLE
Article: Emulating a Raspberry Pi in QEMU

### PR DESCRIPTION
**Pull request by stawiski**

### CODE REVIEW

File: _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## What is QEMU?
+
+[QEMU](https://www.qemu.org/) is an open-source software tool that provides hardware virtualization capabilities. It allows you to run operating systems and programs designed for one architecture on a different architecture, thereby enabling cross-platform emulation. QEMU can emulate various CPUs, including x86, ARM, and more. It can also emulate a variety of devices such as network cards, and storage devices. QEMU is a great tool for testing software on different architectures without having to buy the hardware.
+
+## Environment setup
```
[the remark this answers is not in the document]
Comment (stawiski) [reply _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md:27]: Ok, done

File: _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+To run QEMU we will need the kernel and device tree, so let's copy them out:
+```bash
+$ cp /mnt/image/bcm2710-rpi-3-b-plus.dtb ~
+$ cp /mnt/image/kernel8.img ~
+```
+
+## Setting up SSH
+
+While we have the image mounted, let's sort out SSH as well. Raspberry changed it's default password policy so we no longer can use `pi:raspberry` to log in. We'll need to create new password first. Let's recreate password `rapsberry` using `openssl`:
```
[the remark this answers is not in the document]
Comment (stawiski) [reply _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md]: Ok

File: _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    mcopy /tmp/ssh x:/ && \
+    mcopy /tmp/userconf x:/
+```
+
+That's it!
+
+## Dockerfile
+
+Here's the full Dockerfile, based on Ubuntu 20.04:
+```Dockerfile
```
[the remark this answers is not in the document]
Comment (stawiski) [reply _posts/2023-05-17-emulating-raspberry-pi-in-qemu.md]: Easy!

**Comment by stawiski** [comment]
@bahildebrand @tyhoff Any more comments?

**Comment by stawiski** [comment]
> Sorry for the delay in responding @stawiski . There is no more major feedback from me. Could you move this over to _drafts so that we can schedule it sometime in the future?

Sure. Done!

**Review by bahildebrand — APPROVED** [review]
Nice work!

**Pull request merged**